src/zope/password/password.py

```diff
@@ -450,13 +451,119 @@ def checkPassword(self, encoded_password, password):
         return encoded_password == self.encodePassword(password)[5:]
 
     def match(self, encoded_password):
-        if isinstance(encoded_password, unicode):
+        if isinstance(encoded_password, text_type):
             encoded_password = encoded_password.encode('ascii')
         return (
             encoded_password.startswith(b'{SHA}') or
             encoded_password.startswith(b'{SHA1}'))
 
 
+if bcrypt is not None:
```

Comment:
Conditionally defining the class feels icky.  Instead, could we add an `__init__` which raises an exception if `bcrypt` is None?

```python
    def __init__(self):
        if bcrypt is None:
             raise ValueError("bcrypt is not installed")
        super(BCRYPTPasswordManager, self).__init__()
```